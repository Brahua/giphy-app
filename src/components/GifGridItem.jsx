import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, title, url}) => {
  return (
    <div className="card">
      <img src={url} alt={title} className="card-img-top"/>
      <div className="card-body">
        <h5 className="card-title">{title}</h5>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

export default GifGridItem