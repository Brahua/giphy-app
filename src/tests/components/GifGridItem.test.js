import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from './../../components/GifGridItem';


describe('Tests in <GifGridItem>', () => {
  const gif = {
    id: 'ffsjFDSKBFs456',
    title: 'Title',
    url: 'http://localhost:4200'
  }
  const component = <GifGridItem key={gif.id} {...gif}/>;
  
  let wrapper = shallow(component);

  beforeEach( () => {
    wrapper = shallow(component)
  })


  test('should create the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should show the title of gif', () => {
    const h5 = wrapper.find('h5');
    expect(h5.text().trim()).toBe(gif.title);
  });

  test('should show the img props (src, alt) correctly', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(gif.url);
    expect(img.prop('alt')).toBe(gif.title);
  });

  test('should show animation effect', () => {
    const div = wrapper.find('div')
    const className = (div.getElements()[0]).props.className
    expect(className).toContain('animate__animated animate__bounceIn');
  });
  
})
