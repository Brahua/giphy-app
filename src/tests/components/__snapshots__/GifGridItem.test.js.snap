// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem> should create the component correctly 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Title"
    className="card-img-top"
    src="http://localhost:4200"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Title
    </h5>
  </div>
</div>
`;
